Use mage to setup development environment

neoaggelos opened the pull request:
<!--
Thanks for submitting a pull request. Please fill the template below,
otherwise we will not be able to process this pull request.
-->

#### Summary
<!--
A short summary, referencing related issues:
Closes #0000, References #0000, etc.
-->

Related to #1866 

#### Changes
<!-- What are the changes made in this pull request? -->

- Added mage targets `dev:dbStart`, `dev:dbStop`, `dev:dbErase`, `dev:dbSQL`, `dev:dbRedisCli` and dependencies
- Updated setup instructions in `DEVELOPMENT.md`

#### Notes for Reviewers
<!--
NOTE: This section is optional.

Motivate briefly why it is implemented this way, if that deviates from the
implementation proposal in the referenced issues.
- How should your reviewers approach this pull request?
- @mention reviewers with special requests or questions for them
-->

- Also fixed a few inconsistencies `mage` vs `./mage` in DEVELOPMENT.md
- Should we remove the makefile targets altogether?
- travis still uses `dev.databases.start`. We should update that too in `.travis.yml`

#### Checklist
<!-- Make sure that this pull request is complete. -->

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [x] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [X] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [X] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

htdvisser commented on .mage/dev.go:
```diff
@@ -68,8 +69,72 @@ func (Dev) Misspell() error {
+	return append(devDockerComposeFlags, args...)
+}
+
+func execDC(args ...string) error {
```
Probably better to call this `execDockerCompose` in full, since `execDC` isn't obvious in this package.

htdvisser commented on .mage/dev.go:
```diff
@@ -68,8 +69,72 @@ func (Dev) Misspell() error {
+	if err := execDC(append([]string{"up", "-d"}, devDatabases...)...); err != nil {
+		return err
+	}
+
```
Nit: no need for spacing in small funcs

htdvisser commented on .mage/dev.go:
```diff
@@ -68,8 +69,72 @@ func (Dev) Misspell() error {
+	return err
+}
+
+// DBStart starts the databases of the development environment
```
Comments end with a period.
also in other places

htdvisser commented on .mage/dev.go:
```diff
@@ -68,8 +69,72 @@ func (Dev) Misspell() error {
+
 // InitStack initializes the Stack.
 func (Dev) InitStack() error {
+	mg.Deps(Dev.DBStart)
```
Not everyone uses the databases in docker, so let's not force this.

neoaggelos replied:
Followed previous behaviour, can remove. Will need to update DEVELOPMENT.md instructions accordingly

htdvisser commented on DEVELOPMENT.md:
```diff
@@ -40,25 +40,25 @@ This starts a CockroachDB and Redis database in Docker containers, creates a dat
 # The contents of the databases will be saved in .dev/data.
 
-$ make dev.databases.erase # Stop all databases and erase storage.
+$ ./mage dev:dbErase # Stop all databases and erase storage.
```
Let's make the comment consistent with the ones above while we're making changes.

> Stops all databases and erases storage.

And add periods to the comments above.

neoaggelos commented:
Updated based on comments. Removed old makefile targets.

There still is `make git.diff` that is being used, but that should be done in a separate PR, I think.

Please take a look at the `.travis.yml` change to make sure that we don't break anything.
